packages/yarnpkg-core/sources/scriptUtils.ts

```diff
@@ -150,14 +154,19 @@ export async function prepareExternalProject(cwd: PortablePath, outputPath: Port
       const stdin = null;
       const {stdout, stderr} = configuration.getSubprocessStreams(logFile, {prefix: cwd, report});
 
-      const packageManager = await detectPackageManager(cwd);
+      const projectName: string = await xfs.readFilePromise(ppath.join(cwd, `package.json` as PortablePath), `utf8`)
+        .then((packageInfo: string) => JSON.parse(packageInfo).name);
```

Comment:
Thinking about it, I'm not sure the name makes sense, since if you look into the file you're supposed to already know what package you're inspecting (since it's in the same message as the file path you used to access the file).

Reply:
I don't think I fully understand what you're recommending:
- Is the project name available somewhere else that we have access to here?
- Or would you rather use some other identifier here instead of the name field?

As long as the user is able to figure out what package these error messages are referring to, I'm ok with changing this to something else

Reply:
The callers of `prepareExternalProject` has that information, more specifically they have the `Locator` which contains the information you need

https://github.com/yarnpkg/berry/blob/eeb78deb82904e44ee0f0ed333ca7e7d4b7a8543/packages/plugin-github/sources/GithubFetcher.ts#L39
https://github.com/yarnpkg/berry/blob/724a260c57f2800fd06ad382376c4ecf0f0f0cee/packages/plugin-git/sources/GitFetcher.ts#L51

Reply:
Thanks! Updated